test/jdk/java/foreign/TestSlices.java

```diff
@@ -128,7 +128,7 @@ public void testUnboundedSlice() {
             MemorySegment segment = arena.allocate(MemoryLayout.sequenceLayout(2, ValueLayout.JAVA_INT)); // size = 8
             MemorySegment slice = segment.asSlice(0, ValueLayout.JAVA_INT); // size = 4
             assertThrows(IndexOutOfBoundsException.class, () -> slice.getAtIndex(ValueLayout.JAVA_INT, 1));
-            MemorySegment unbounded = slice.asUnbounded();
+            MemorySegment unbounded = slice.reinterpret(Long.MAX_VALUE, arena.scope(), null);
```

Comment:
Looking at this, I'm wondering if the whole test `testUnboundedSlice` is still needed at all, now that `asUnbounded` is removed. I'd say just remove the test.